test/gtest/ucp/test_ucp_request.cc

```diff
@@ -452,7 +452,8 @@ class test_proto_reset : public ucp_test {
     {
         const ucp_datatype_iter_t *dt_iter = &req->send.state.dt_iter;
 
-        return !ucp_datatype_iter_is_begin(dt_iter) &&
+        return ((req->flags & UCP_REQUEST_FLAG_MIDDLE_FRAGMENT) ||
```

Comment:
use `ucp_proto_multi_is_middle_fragment`?

Reply:
Yes, looks logical, it was my initial impl, but then I removed it, because including "proto_multi.inl" into C++ translation unit requires quite some unrelated changes in that file: see my reverted commit which fixes some of them: https://github.com/openucx/ucx/pull/10263/commits/e74c74ac6d39d5bed8dd3d019fc7cb71ca82a55a

So I thought it's not worth it.
Else we can move `ucp_proto_multi_is_middle_fragment` function to `proto_multi.h`